### PR TITLE
Rely on the service_provider fact

### PR DESCRIPTION
**Pull request by ekohl**
This avoids hardcoding the service provider and greatly simplifies the code.

### CODE REVIEW

**Comment by mmoll** [comment]
@ekohl tests fail :broken_heart:

**Comment by mmoll** [comment]
is this a 12.0.0 candidate?

**Comment by ekohl** [comment]
No, it's a branch I had locally that I thought I should finish at some point. I'd rather get 12.0.0 out now and merge this later.

**Comment by mmoll** [comment]
what's the status here?

**Comment by ekohl** [comment]
Rebased and updated the tests

**Comment by mmoll** [comment]
test failures related?

**Comment by ekohl** [comment]
I'm not entirely sure how to explain the test failures. It's also annoying we don't have EL7 tests now so that should be fixed first by using an older image.

**Comment by ekohl** [comment]
So that does appear to be the problem:
```
2021-03-29T15:04:07.1177819Z debian9-64.example.com 15:04:07$ facter --json "service_provider"
2021-03-29T15:04:07.2509850Z   {
2021-03-29T15:04:07.2510867Z     "service_provider": ""
2021-03-29T15:04:07.2517137Z   }
```

**Comment by ekohl** [comment]
Thinking more about this, it may well be that `facter` doesn't include the custom Puppet facts.

**Comment by ekohl** [comment]
After such a long time I finally found out the problem and it was really stupid: I used `/usr/bin/systemctl` but on Debian it's `/bin/systemctl`. Somehow Puppet didn't raise an error about the command not being found and it fell back to a real restart instead of a reload. Let's see if CI agrees.

**Comment by ekohl** [comment]
This finally passes. The other failures are unrelated.

File: manifests/params.pp [hunk 1]
```diff
@@ -288,37 +288,30 @@
   $systemd_unit_name = 'puppet-run'
   # Mechanisms to manage and reload/restart the agent
   # If supported on the OS, reloading is prefered since it does not kill a currently active puppet run
-  case $facts['os']['family'] {
-    'Debian' : {
-      $agent_restart_command = "/usr/sbin/service ${service_name} reload"
-      $unavailable_runmodes = []
+  if $facts['service_provider'] == 'systemd' {
+    $agent_restart_command = "/bin/systemctl reload-or-restart ${service_name}"
```
Comment (wbclark) [inline manifests/params.pp:292]: Nice, TIL

**Review by wbclark — COMMENTED** [review]
What if any is the relationship between these two commits? Is the change to `server_jvm_max_heap_size` better suited to a separate PR?

**Comment by ekohl** [comment]
It's something I found while testing this manually. Technically it could be a separate PR, but I think there's little value in that. It is only for the acceptance tests so it's not like users need to know about it via the changelog. Combining them saves CI resources.

**Comment by wbclark** [comment]
> It's something I found while testing this manually. Technically it could be a separate PR, but I think there's little value in that. It is only for the acceptance tests so it's not like users need to know about it via the changelog. Combining them saves CI resources.

Ah, got it, TY

**Review by wbclark — APPROVED** [review]

**Pull request merged**